src/box/sql/parse_def.c

```diff
@@ -235,3 +244,52 @@ sql_parse_add_unique(struct Parse *parse, struct SrcList *table_name,
 	parse->src_list = table_name;
 	unique_list_append(parse, name, cols);
 }
+
+/** Fill in the description of the PRIMARY KEY. */
+static void
+primary_key_fill(struct Parse *parse, const struct Token *name,
+		 struct ExprList *cols)
+{
+	if (parse->primary_key.cols != NULL) {
+		const char *space_name;
+		if (parse->src_list == NULL)
+			space_name = new_table_name(parse);
+		else
+			space_name = parse->src_list->a[0].zName;
+		diag_set(ClientError, ER_CREATE_SPACE, space_name,
+			 "primary key has been already declared");
+		parse->is_aborted = true;
+		sql_expr_list_delete(cols);
+		sql_expr_list_delete(parse->primary_key.cols);
```

Comment:
I suggest to either free this (parse->primary_key) in one place - when Parse is being destroyed. Or at least nullify this member here after you freed it. To ensure it won't be used after free.